x-pack/plugins/fleet/public/components/agentless_enrollment_flyout/index.tsx

```diff
@@ -0,0 +1,213 @@
+  packagePolicy: PackagePolicy;
+  agentPolicy?: AgentPolicy;
+}) => {
+  const core = useStartServices();
+  const { notifications } = core;
+  const [confirmEnrollmentStatus, setConfirmEnrollmentStatus] = useState<EuiStepStatus>('loading');
+  const [confirmDataStatus, setConfirmDataStatus] = useState<EuiStepStatus>('disabled');
+  const [agentData, setAgentData] = useState<Agent>();
+
+  // Clear agent data polling
+  // Called when component is unmounted or when agent is healthy
+  const agentDataInterval = useRef<NodeJS.Timeout>();
+  const clearAgentDataPolling = useMemo(() => {
+    return () => {
+      if (agentDataInterval.current) {
+        clearInterval(agentDataInterval.current);
+      }
+    };
+  }, [agentDataInterval]);
+
+  // Fetch agent(s) data for the first associated agent policy
+  // Polls every 30 seconds until agent is found and healthy
+  useEffect(() => {
+    const fetchAgents = async () => {
+      const { data: agentsData, error } = await sendGetAgents({
+        kuery: `${AGENTS_PREFIX}.policy_id: "${packagePolicy.policy_ids[0]}"`,
+      });
+
+      if (error) {
+        notifications.toasts.addError(error, {
+          title: i18n.translate(
+            'xpack.fleet.epm.packageDetails.integrationList.agentlessStatusError',
+            {
+              defaultMessage: 'Error fetching agentless status information',
+            }
+          ),
+        });
+      }
+
+      if (agentsData?.items?.[0]) {
+        setAgentData(agentsData.items?.[0]);
+      }
+    };
+
+    fetchAgents();
+    agentDataInterval.current = setInterval(() => {
+      fetchAgents();
+    }, REFRESH_INTERVAL_MS);
+
+    return () => clearAgentDataPolling();
+  }, [clearAgentDataPolling, notifications.toasts, packagePolicy.policy_ids]);
+
+  // Watches agent data and updates step statuses and clears polling when agent is healthy
+  useEffect(() => {
+    if (agentData) {
+      if (agentData.status === 'online') {
+        setConfirmEnrollmentStatus('complete');
+        setConfirmDataStatus('loading');
+        clearAgentDataPolling();
+      } else if (agentData.status === 'error' || agentData.status === 'degraded') {
+        setConfirmEnrollmentStatus('danger');
+        setConfirmDataStatus('disabled');
+      } else {
+        setConfirmEnrollmentStatus('loading');
+        setConfirmDataStatus('disabled');
+      }
+    } else {
+      setConfirmEnrollmentStatus('loading');
+      setConfirmDataStatus('disabled');
+    }
+  }, [agentData, clearAgentDataPolling]);
+
+  // Calculate integration title from the base package info and what
+  // is configured on the package policy.
+  const { data: packageInfoData } = useGetPackageInfoByKeyQuery(
+    packagePolicy.package!.name,
+    packagePolicy.package!.version,
+    {
+      prerelease: true,
+    }
+  );
+  const [packageInfo, setPackageInfo] = useState<PackageInfo>();
+  const [integrationTitle, setIntegrationTitle] = useState<string>(packagePolicy.name);
+  useEffect(() => {
```

Comment:
nit: I think we can remove this `useEffect` and refactor `integrationTitle` call a function instead of using `state`